### PR TITLE
Feature/book update soft delete

### PR DESCRIPTION
**Pull request by Cayman4010**

### CODE REVIEW

**Review by dmytrocherepov — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -52,18 +54,34 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.mapstruct</groupId>
+            <artifactId>mapstruct</artifactId>
+            <version>${org.mapstruct.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.liquibase</groupId>
+            <artifactId>liquibase-core</artifactId>
+            <version>${liquibase.version}</version>
```
Comment (Elena-Bruyako) [inline pom.xml]: Read about properties in pom files. 
You didn't declare the alias of liquibase.version in properties, so you are using not existing version. 

Also, remove the version here as spring-boot-starter-parent already includes liquibase

File: pom.xml [hunk 2]
```diff
@@ -78,6 +96,33 @@
...
                     <linkXRef>false</linkXRef>
+                    <sourceDirectories>src</sourceDirectories>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.11.0</version>
+                <configuration>
+                    <source>11</source>
```
Comment (Elena-Bruyako) [inline pom.xml]: You are using the 17 java version. Why 11?

File: pom.xml [hunk 3]
```diff
@@ -78,6 +96,33 @@
...
+                    <target>11</target>
```
Comment (Elena-Bruyako) [inline pom.xml]: The same

File: src/main/java/bookstore/controller/BookController.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    public BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
+        return bookService.save(requestDto);
+    }
+
+    @GetMapping("/{id}")
+    public BookDto getBookById(@PathVariable Long id) {
+        return bookService.getById(id);
+    }
+
+    @ResponseStatus(HttpStatus.NO_CONTENT)
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/controller/BookController.java:43]: If you add @ResponseStatus(HttpStatus.NO_CONTENT) for one endpoint,  you should add it for all other endpoints

File: src/main/java/bookstore/mapper/BookMapper.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+@Mapper(componentModel = "spring",
+        injectionStrategy = InjectionStrategy.CONSTRUCTOR,
+        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
+        implementationPackage = "<PACKAGE_NAME>.impl")
+public interface BookMapper {
+    BookDto toDto(Book book);
+
+    Book toBook(CreateBookRequestDto requestDto);
+
+    void updateBookFromDto(CreateBookRequestDto requestDto, @MappingTarget Book book);
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/mapper/BookMapper.java]:
```suggestion
    void updateBook(CreateBookRequestDto requestDto, @MappingTarget Book book);
```

What if CreateBookRequestDto includes null values? Your book will be updated with null? Read about NullValuePropertyMappingStrategy.IGNORE)

File: src/main/java/bookstore/repository/BookRepository.java [hunk 6]
```diff
@@ -1,10 +1,7 @@
...
 
 import bookstore.model.Book;
-import java.util.List;
+import org.springframework.data.jpa.repository.JpaRepository;
 
-public interface BookRepository {
-    Book save(Book book);
-
-    List<Book> findAll();
+public interface BookRepository extends JpaRepository<Book, Long> {
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/repository/BookRepository.java:8]: Annotation Repository is missing

File: src/main/java/bookstore/service/impl/BookServiceImpl.java [hunk 7]
```diff
@@ -1,26 +1,57 @@
...
-        return bookRepository.save(book);
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public BookDto updateById(Long id, CreateBookRequestDto requestDto) {
+        Optional<Book> optionalBook = bookRepository.findById(id);
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/service/impl/BookServiceImpl.java]: Let's simplify this code by using bookRepository.findById(id).orElseThrow()....

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution!

File: pom.xml (hunk 1 above)
Comment (NazarSmith228) [reply pom.xml]: not fixed

File: pom.xml [hunk 8]
```diff
@@ -52,18 +54,34 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.liquibase</groupId>
+            <artifactId>liquibase-core</artifactId>
+            <version>${liquibase.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.liquibase</groupId>
+            <artifactId>liquibase-maven-plugin</artifactId>
+            <version>${liquibase.version}</version>
```
Comment (NazarSmith228) [inline pom.xml]: this is a plugin - not a dependency

pls remove it from `dependencies` section

File: src/main/java/bookstore/dto/BookDto.java [hunk 9]
```diff
@@ -0,0 +1,7 @@
+package bookstore.dto;
+
+import java.math.BigDecimal;
+
+public record BookDto(Long id, String title, String author, String isbn,
```
Comment (NazarSmith228) [inline src/main/java/bookstore/dto/BookDto.java]: a general recommendation on record formatting

if your record has 1-2 fields, you can:
```
public record SomeRecord(String field1, Integer field2) {
```
if your record has more than 2 fields:
```
public record SomeRecord(String field1, 
                         int field2,
                         Double field3,
                         boolean field4) {

```
or alrternative:
```
public record SomeRecord(
         String field1, 
         int field2,
         Double field3,
         boolean field4) {
```

**Review by NazarSmith228 — CHANGES_REQUESTED** [review]
Good job! Just some minor remarks to fix

**Review by Elena-Bruyako — APPROVED** [review]
Well done!

**Pull request merged**